src/components/contact-us/contact-us-controller.ts

```diff
@@ -111,6 +118,30 @@ export function contactUsGet(req: Request, res: Response): void {
   return res.render("contact-us/index-public-contact-us.njk", options);
 }
 
+export function contactUsGetFromTriagePage(req: Request, res: Response): void {
+  const queryParams = new URLSearchParams({
+    ...(validateAppId(req.query.appSessionId as string) && {
```

Comment:
what do the '...' do here?

Reply:
Thanks @dbes-gds - it uses [spread syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#conditionally_adding_properties_to_an_object) to conditionally add a property to the object literal that's passed to `URLSearchParams` where left hand side of the `&&` is truthy.

```typescript
  const queryParams = new URLSearchParams({
    ...(validateAppId(req.query.appSessionId as string) && {
      appSessionId: getAppSessionId(req.query.appSessionId as string),
    }),
    ...(getAppErrorCode(req.query.appErrorCode as string) && {
      appErrorCode: getAppErrorCode(req.query.appErrorCode as string),
    }),
    ...(validateReferer(req.query.fromURL as string) && {
      fromURL: validateReferer(req.query.fromURL as string),
    }),
  });
```

My main motivation was to avoid empty values appearing as search params in the query string - but this might not be the best way to code it (I went for it because it was more concise). I think the same could be achieved with code along these lines, but it seemed quite long:

```typescript
  const params = {};
  
  if(validateAppId(req.query.appSessionId as string)) {
    params.appSessionId = getAppSessionId(req.query.appSessionId as string)
  }

  if(getAppErrorCode(req.query.appErrorCode as string)) {
    params.appSessionId = getAppErrorCode(req.query.appErrorCode as string)
  }

  if(validateReferer(req.query.fromURL as string)) {
    params.fromURL = validateReferer(req.query.fromURL as string)
  }

  const queryParams = new URLSearchParams(params);
```

I'd be very happy to change the implementation to be along these lines though (or another way) 👍🏽

Reply:
@gtvj fine with this if it does the trick, just wasn't used to the syntax